safe/utilities/analysis.py

```diff
@@ -391,9 +378,31 @@ def get_clip_parameters(self):
             exposure_layer.extent(),
             exposure_layer.crs())
 
-        # Reproject all extents to EPSG:4326 if needed
-        geo_crs = QgsCoordinateReferenceSystem()
-        geo_crs.createFromId(4326, QgsCoordinateReferenceSystem.EpsgCrsId)
+        # get the current view extents
+        viewport_extent = viewport_geo_array(self.map_canvas)
+
+        # Set the anlysis extents based on user's desired behaviour
+        settings = QtCore.QSettings()
+        mode_name = settings.value(
+            'inasafe/analysis_extents_mode',
+            'HazardExposureView')
+        # Default to using canvas extents if no case below matches
+        analysis_geoextent = viewport_extent
+        if mode_name == 'HazardExposureView':
+            analysis_geoextent = viewport_extent
+
+        elif mode_name == 'HazardExposure':
+            analysis_geoextent = None
+
+        elif mode_name == 'HazardExposureBookmark' or \
```

Comment:
Which one is better? Using bracket or backslash for something like this?

My personal taste : don't use backslash, if possible.